src/main/java/co/cask/hydrator/salesforce/authenticator/Authenticator.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.authenticator;
+
+import co.cask.hydrator.salesforce.SalesforceUtils;
+import com.google.gson.Gson;
+import com.sforce.ws.ConnectorConfig;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * Authentication to Salesforce via oauth2
+ */
+public class Authenticator {
+  private static final Gson GSON = new Gson();
+
+  /**
+   * Authenticates via oauth2 to salesforce and returns a connectorConfig
+   * which can be used by salesforce libraries to make a connection.
+   *
+   * @return ConnectorConfig which can be used to create BulkConnection and PartnerConnection
+   */
+  public static ConnectorConfig createConnectorConfig(AuthenticatorCredentials credentials) {
+    try {
+      AuthResponse authResponse = oauthLogin(credentials);
+      ConnectorConfig connectorConfig = new ConnectorConfig();
+      connectorConfig.setSessionId(authResponse.getAccessToken());
+      // https://instance_name—api.salesforce.com/services/async/APIversion/job/jobid/batch
+      String apiVersion = SalesforceUtils.API_VERSION;
+      String restEndpoint = String.format("%s/services/async/%s", authResponse.getInstanceUrl(), apiVersion);
+      String serviceEndPoint = String.format("%s/services/Soap/u/%s", authResponse.getInstanceUrl(), apiVersion);
+      connectorConfig.setRestEndpoint(restEndpoint);
+      connectorConfig.setServiceEndpoint(serviceEndPoint);
+      // This should only be false when doing debugging.
+      connectorConfig.setCompression(true);
+      // Set this to true to see HTTP requests and responses on stdout
+      connectorConfig.setTraceMessage(false);
+
+      return connectorConfig;
+    } catch (Exception e) {
+      throw new RuntimeException("Connection to salesforce with plugin configurations failed", e);
+    }
+  }
+
+  /**
+   * Authenticate via oauth2 to salesforce and return response to auth request.
+   *
+   * @return AuthResponse response to http request
+   */
+  public static AuthResponse oauthLogin(AuthenticatorCredentials credentials) throws Exception {
+    SslContextFactory sslContextFactory = new SslContextFactory();
+    HttpClient httpClient = new HttpClient(sslContextFactory);
+    try {
+      httpClient.start();
+      String response = httpClient.POST(credentials.getLoginUrl()).param("grant_type", "password")
+        .param("client_id", credentials.getClientId())
+        .param("client_secret", credentials.getClientSecret())
+        .param("username", credentials.getUsername())
+        .param("password", credentials.getPassword()).send().getContentAsString();
```

Comment:
Check if received response should be closed as well

Reply:
https://www.eclipse.org/jetty/javadoc/current/org/eclipse/jetty/client/api/ContentResponse.html

Seems like it does not need to be closed